account/email/verification_service.go

```diff
@@ -0,0 +1,95 @@
+package email
+
+import "github.com/fabric8-services/fabric8-auth/application"
+import "github.com/fabric8-services/fabric8-auth/account"
+import "github.com/fabric8-services/fabric8-auth/log"
+import "github.com/fabric8-services/fabric8-auth/errors"
+import errs "github.com/pkg/errors"
+import "github.com/satori/go.uuid"
+
+import "context"
+
+type EmailVerificationService interface {
+	SendVerificationCode(ctx context.Context, user account.User) error
+	VerifyCode(ctx context.Context, code string) (bool, error)
+}
+
+type EmailVerificationClient struct {
+	db application.DB
+}
+
+// NewKeycloakIDPServiceClient creates a new Keycloakc
+func NewEmailVerificationClient(db application.DB) *EmailVerificationClient {
+	return &EmailVerificationClient{
+		db: db,
+	}
+}
+
+// SendVerificationCode generates and sends out an email with verification code.
+func (c *EmailVerificationClient) SendVerificationCode(ctx context.Context, user account.User) error {
+
+	newVerificationCode := account.VerificationCode{
+		User:     user,
+		Code:     uuid.NewV4().String(),
+		Verified: false,
+	}
+
+	return application.Transactional(c.db, func(appl application.Application) error {
+		log.Info(ctx, map[string]interface{}{
+			"email": user.Email,
+		}, "verification code to be sent")
+		err := appl.VerificationCodes().Create(ctx, &newVerificationCode)
+		/*
+			TODO: Invoke the EmailService to send out an email
+		*/
+		return err
+	})
+	return nil
+}
+
+// VerifyCode validates whether the code is present in our database.
+func (c *EmailVerificationClient) VerifyCode(ctx context.Context, code string) (bool, error) {
```

Comment:
I think VerifyEmail() would be a better name. This method should:
- Load the code
- If found, set user.email_verfied = true
- Hard delete the code